### ISSUE TITLE
package friendly design

### ISSUE THREAD

**Issue by cederom** [MISP/MISP#1508]
Hello,

I am working on FreeBSD port of MISP. I like FreeBSD because unlike Linux its self-compatible, minimal and easy to maintain. Everything worked fine with 2.4.48. Then things changed and with 2.4.51 port does not build anymore. Sumbodules and Composer dependencies changed in a way they are impossible to track that would preserve GIT structure. GIT structure on the other hand is important because of fast changes and updates.

Please work out a design that would not imply internal dependencies and become more package friendly. This would be very important with MISP 3. To get the package and dependencies that are easy to deploy. Noone is really interested in all those setup steps that quickly change over time and highly depends on the underlying OS..

Please consider more package friendly design of future MISP releases.

Thank you :-)

**Comment by rotanid** [comment]
Hi, thanks for your feedback.
Which changes are you referring to exactly?
from 2.4.48 to 2.4.51 there was no change at all to composer.json and only a bugfix in the .gitmodules and some regular updates to the submodules.
Aside from that, do you have some more specific suggestions for us?

as an open source project we always welcome contributions via pull requests, too. (if they are reasonable for everybody)

**Comment by cederom** [comment]
Long story short there are two ways to maintain a software - manual build via git or binary packages with no git. I have tried to create a MISP package with git files inside but this does not seem to work well. Thus my question on future plans on a release organization..

After updating 2.4.48 to 2.4.51 some files seems to change and simple reabuild did not work, git files changed in submodules (even though I stayed with 2.4.48 at first?). But that is probably my local problem with creating a package. After removing all old files, updating to 2.4.51, and working on a clean system it works now again. Still, if I want to build a package, I need a deterministic static list of files to be installed. This will change with updates, maybe even with each git pull. Also the port may use packages/submodules that are already present on the system. As for now I create a package with all-in-one that is MISP and python/php submodules.. it does not seem to work well.

I understand putting submodules may be easy to simply git clone type of manual install. But when you want to automate things and build a package, then submodules are usually provided by the OS packaging system as separate packages.. and all those packages consist of deterministic static list of files.

This is my question and request if it is possible to consider 'package' approach in future releases..? Or you plan to stay with git+submodules and updates via git pull instead system packages update? If the second option is your choice then there is no point in creating a package.. just want to make sure :-)

Thanks for quick answer :-)
